packages/json/src/node/index.ts

```diff
@@ -1,7 +1,22 @@
-import type { DeepReadonly, ItemNode, PairNode } from '@spyglassmc/core'
 import * as core from '@spyglassmc/core'
+import type * as mcdoc from '@spyglassmc/mcdoc'
 import { JsonStringOptions } from '../parser/index.js'
 
+export interface JsonFileNode extends core.AstNode {
+	readonly type: 'json:file'
+	readonly children: [JsonNode]
+}
+export namespace JsonFileNode {
+	/* istanbul ignore next */
+	export function is(obj: object | undefined): obj is JsonFileNode {
+		return (obj as JsonFileNode | undefined)?.type === 'json:file'
+	}
+}
+
+interface JsonBaseNode {
+	typeDef?: mcdoc.McdocType
```

Comment:
This should be a SimplifiedMcdocType

Reply:
I sort of prefer working with `McdocType` everywere and think the simplified types should be left mostly internal to the checker. We can still just choose not to handle certain kinds in for example the completer.

Reply:
I definitely prefer types that show exactly what kind of data you can expect, no more no less.

I intentionally provide a simplified mcdoctype in the interface of attachTypeInfo, for exactly this reason, to tell the user that this is the only kind of types that it can expect.